apps/docs/content/guides/functions.mdx

```diff
@@ -14,6 +14,40 @@ Edge Functions are server-side TypeScript functions, distributed globally at the
 - It is TypeScript first and supports WASM.
 - Edge Functions are globally distributed for low-latency.
 
+## How it works
+
+- **Request enters an edge gateway (relay)** — the gateway routes traffic, handles auth headers/JWT validation, and applies routing/traffic rules.
+- **Auth & policies are applied** — the gateway (or your function) can validate Supabase JWTs, apply rate-limits, and centralize security checks before executing code.
+- **Deno edge runtime executes your function** — the function runs on a regionally-distributed Deno node closest to the user for minimal latency.
+- **Integrations & data access** — functions commonly call Supabase APIs (Auth, Postgres, Storage) or third-party APIs. For Postgres, prefer connection strategies suited for edge/serverless environments (see the `connect-to-postgres` guide).
+- **Observability and logs** — invocations emit logs and metrics you can explore in the dashboard or downstream monitoring (Sentry, etc.).
+- **Response returns via the gateway** — the gateway forwards the response back to the client and records request metadata.
+
+## Quick technical notes
+
+- **Runtime:** Deno (TypeScript-first). Functions are simple `.ts` files that export a handler.
+- **Local dev parity:** `supabase functions serve` provides a local runtime similar to production for faster iteration.
```

Comment:
Local dev parity: Use Supabase CLI for a local runtime similar to production for faster iteration (`supabase functions serve` command)